Recipes/CamelCasePropertyNamesWithConstructorParametersRequiredContractResolver/CamelCasePropertyNamesWithConstructorParametersRequiredContractResolver.cs

```diff
@@ -0,0 +1,83 @@
+﻿// --------------------------------------------------------------------------------------------------------------------
+// <copyright file="ConstructorParametersRequiredContractResolver.cs">
+//     Copyright (c) 2016. All rights reserved. Licensed under the MIT license. See LICENSE file in
+//     the project root for full license information.
+// </copyright>
+// <auto-generated>
+// Sourced from NuGet package. Will be overwritten with package update except in Spritely.Recipes source.
+// </auto-generated>
+// --------------------------------------------------------------------------------------------------------------------
+
+namespace Spritely.Recipes
+{
+    using System;
+    using System.Reflection;
+
+    using Newtonsoft.Json;
+    using Newtonsoft.Json.Serialization;
+
+    /// <summary>
+    /// Resolves member mappings for a type using camel casing property names.
+    /// Also requires that constructor parameters are defined in the json string
+    /// when deserializing, for types with non-default constructors. 
+    /// </summary>
+    /// <remarks>
+    /// See <a href="https://stackoverflow.com/questions/37416233/json-net-should-not-use-default-values-for-constructor-parameters-should-use-de"/>
+    /// </remarks>
+#if !RecipesProject
+    [System.Diagnostics.DebuggerStepThrough]
+    [System.Diagnostics.CodeAnalysis.ExcludeFromCodeCoverage]
+    [System.CodeDom.Compiler.GeneratedCode("Spritely.Recipes", "See package version number")]
+#endif
+    public class CamelCasePropertyNamesWithConstructorParametersRequiredContractResolver
+        : CamelCasePropertyNamesContractResolver
+    {
+        /// <summary>
+        /// As of 7.0.1, Json.NET suggests using a static instance for "stateless" contract resolvers, for performance reasons.
+        /// <a href="http://www.newtonsoft.com/json/help/html/ContractResolver.htm"/>
+        /// <a href="http://www.newtonsoft.com/json/help/html/M_Newtonsoft_Json_Serialization_DefaultContractResolver__ctor_1.htm"/>
+        /// "Use the parameterless constructor and cache instances of the contract resolver within your application for optimal performance."
+        /// Also, <a href="https://stackoverflow.com/questions/33557737/does-json-net-cache-types-serialization-information"/>
+        /// </summary>
+        public static CamelCasePropertyNamesWithConstructorParametersRequiredContractResolver Instance
+            => new CamelCasePropertyNamesWithConstructorParametersRequiredContractResolver();
```

Comment:
As much as I love C# 6 syntax, I avoid it in recipes since it gets source included in other projects and I don't know what they will be using to build. Can you go back to old syntax on non-test projects?